plugins/elantp/fu-elantp-hid-device.c

```diff
@@ -192,19 +237,77 @@ fu_elantp_hid_device_read_hatpic_enable(FuElantpHidDevice *self, GError **error)
 	return TRUE;
 }
 
+static gboolean
+fu_elantp_hid_device_get_forcetable_address(FuElantpHidDevice *self, GError **error)
+{
+	guint8 buf[2] = {0x0};
+	guint16 addr_wrds;
+
+	if (self->iap_ver == 0x3) {
+		self->force_table_addr = 0xFF40 * 2;
+		return TRUE;
+	}
+	if (!fu_elantp_hid_device_read_cmd(self, ETP_CMD_FORCE_ADDR, buf, sizeof(buf), error)) {
+		g_prefix_error(error, "failed to read force table address cmd: ");
+		return FALSE;
+	}
+	addr_wrds = fu_memread_uint16(buf, G_LITTLE_ENDIAN);
+	if (addr_wrds % 32 != 0) {
+		g_prefix_error(error, "illgeal force table address (%x): ", addr_wrds);
+		return FALSE;
+	}
+
+	self->force_table_addr = addr_wrds * 2;
+
+	/* success */
+	return TRUE;
+}
+
+static gboolean
+fu_elantp_hid_device_write_fw_password(FuElantpHidDevice *self,
+				       guint16 ic_type,
+				       guint16 iap_ver,
+				       GError **error)
+{
+	guint8 buf[2] = {0x0};
+	guint16 pw;
+	guint16 value;
+	if (iap_ver == 0x5 && ic_type == 0x13)
+		pw = ETP_I2C_IC13_IAPV5_PW;
+	else
+		return TRUE;
```

Comment:
can you reverse this please, i.e. only return when it's not 0x5 and 0x13 -- then you can set `guint16 pw = ETP_I2C_IC13_IAPV5_PW`